components/tool/taskManager/src/domain/test/task/FinishWorkTaskUseCaseSpec.js

```diff
@@ -0,0 +1,66 @@
+/* eslint react/jsx-no-undef:0 */
+/* eslint no-undef:0 */
+import chai, {expect} from 'chai'
+import sinon from 'sinon'
+import sinonChai from 'sinon-chai'
+
+import Domain from '../../index.js'
+
+chai.use(sinonChai)
+
+describe('[Domain] FinishWorkTaskUseCase', () => {
+  const domain = new Domain()
+  const config = domain.get('config')
+  const runSimpleTaskUseCase = domain.get('run_simple_task_use_case')
+  const finishWorkTaskUseCase = domain.get('finish_work_task_use_case')
+
+  it('should update the work internal information when it finishes', async () => {
+    // Given
+    const localState = await runSimpleTaskUseCase.execute({
+      localState: {
+        tasks: []
+      },
+      name: 'Simple task with one work',
+      start: () => null
+    })
+
+    // When
+    const stateResult = await finishWorkTaskUseCase.execute({
+      localState,
+      workId: localState.tasks[0].work[0].id,
+      taskId: localState.tasks[0].id
+    })
+
+    // Then
+    const task = stateResult.tasks[0]
+    const work = task.work[0]
+    expect(work.status).to.eql(config.get('AVAILABLE_STATUS').COMPLETED)
```

Comment:
🍆 I might be a bit late on this but perhaps `job` would be a better name for what this is supposed to mean 😅